envisage/_entry_point.py

```diff
@@ -0,0 +1,36 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" This module provides functions to be advertised in the distribution
+entry points.
+"""
+
+import pkg_resources
+
+
+def info(request):
+    """ Return a configuration for contributing examples to the
+    Demo application.
+    Parameters
+    ----------
+    request : dict
+        Information provided by the demo application.
+        Currently this is a placeholder.
+    Returns
+    -------
+    response : dict
+    """
+    return dict(
+        version=1,
+        name="Envisage Examples",
+        root=(
+            pkg_resources.resource_filename("envisage", "examples/demo")
```

Comment:
Given the package data is now defined under `envisage.examples`, it would be more correct to say `pkg_resources.resource_filename("envisage.examples", "demo")`. Though, this would not fail, I think, as `pkg_resources` is not going to complain.

Could you add a test for this little function please?

(I think in the future we will migrate this over to importlib, but that's probably best dealt with separately with other current uses of `pkg_resources`).

Reply:
ah I remember now making similar changes in traits: https://github.com/enthought/traits/pull/1348

I will add a similar test to what was added in that PR, and I think I will also move and rename this `_entry_point.py` file so that its purpose is more clear.